retriever/v2/server.go

```diff
@@ -84,8 +84,11 @@ func (s *Server) RetrieveBlob(ctx context.Context, req *pb.BlobRequest) (*pb.Blo
 	if err != nil {
 		return nil, err
 	}
-	restored := bytes.TrimRight(data, "\x00")
-	restored = codec.RemoveEmptyByteFromPaddedBytes(restored)
+
+	restored, err := codec.CheckAndRemoveInternalPadding(data)
```

Comment:
The old code was wrong, and so is this new code.

This method allegedly retrieves a blob, but this method is partially (and incorrectly) doing blob -> related conversions.

IMO we should just delete the whole `v2` package within `retriever`, if we cannot find a compelling reason it needs to exist